lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+	attr_accessor :cells
+
+	def reset!
+		@cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+	end
+
+	def initialize
+		reset!
+	end
+
+	def display
+		puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+		puts "-----------"
+		puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+		puts "-----------"
+		puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+	end
+
+	def position(user_input)
+		cells[user_input.to_i - 1]
+	end
+
+	def full?
+	   @cells.all? do |position|
+	   position == "X" || position == "O"
+	   end
+	end
+
+	def turn_count
+	   counter = 0
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```